library/src/window_impl.cxx

```diff
@@ -44,6 +55,78 @@ class window_impl::internals
     return this->CachePath;
   }
 
+#if _WIN32
+  /**
+   * Helper function to detect if the
+   * Windows Build Number is equal or greater to a number
+   */
+  bool IsWindowsBuildNumberOrGreater(int buildNumber)
+  {
+    std::string value{};
+    bool result = vtksys::SystemTools::ReadRegistryValue(
+      "HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Windows NT\\CurrentVersion;CurrentBuildNumber",
+      value);
+
+    if (result == true)
+    {
+      try
+      {
+        return std::stoi(value) >= buildNumber;
+      }
+      catch (const std::invalid_argument& e)
+      {
+        f3d::log::debug("Error parsing CurrentBuildNumber", e.what());
+      }
+    }
+    else
+    {
+      f3d::log::debug("Error opening registry key.");
+    }
+
+    return false;
+  }
+
+  /**
+   * Helper function to fetch a DWORD from windows registry.
+   *
+   * @param hKey A handle to an open registry key
+   * @param subKey The path of registry key relative to 'hKey'
+   * @param value The name of the registry value
+   * @param dWord Variable to store the result in
+   */
+  bool ReadRegistryDWord(
+    HKEY hKey, const std::wstring& subKey, const std::wstring& value, DWORD& dWord)
+  {
+    DWORD dataSize = sizeof(DWORD);
+    LONG result = RegGetValueW(
```

Comment:
Can't you use the vtksys wrapper here?

Reply:
I tried it only works if the registry value is a string type not for DWORD

Reply:
Ok